core/storage/classifier/gae_models.py

```diff
@@ -174,3 +174,84 @@ def create(
 
         training_job_instance.put()
         return instance_id
+
+
+class ClassifierExplorationMappingModel(base_models.BaseModel):
+    """Model for mapping exploration attributes to a ClassifierDataModel.
+
+    The id of instances of this class has the form
+    {{exp_id}}.{{exp_version}}.{{state_name}}
+    """
+
+    # The ID of the classifier corresponding to the exploration attributes.
+    classifier_id = ndb.StringProperty(required=True, indexed=True)
+
+    @classmethod
+    def _generate_id(cls, exp_id, exp_version, state_name):
+        """Generates a unique ID for the Classifier Exploration Mapping of the
+        form {{exp_id}}.{{exp_version}}.{{state_name}}
+
+        Args:
+            exp_id: str. ID of the exploration.
+            exp_version: int. The exploration version at the time
+                this training job was created.
+            state_name: str. The name of the state to which the classifier
+                belongs.
+
+        Returns:
+            ID of the new Classifier Exploration Mapping instance.
+        """
+        if isinstance(state_name, unicode):
```

Comment:
@seanlip Yep. Agreed. I'll make a method in utils for this. Oh, I am not sure yet if GAE can store unicode (from what I looked up, it seems to be str), but we aren't storing it as unicode either. If we get a unicode string, we encode it into utf-8. So, this makes the unicode string a python string with these unique characters and can be stored in the ID as required. So, the GAE IDs in this case are all str.